messaging/usecase/message_usecase.go

```diff
@@ -52,33 +52,33 @@ func (u *messageUseCase) SaveMessage(ctx context.Context, message *domain.Messag
 	return nil
 }
 
-func (u *messageUseCase) EditMessage(ctx context.Context, roomID string, userID string, timeStamp time.Time, message string) error {
+func (u *messageUseCase) EditMessage(ctx context.Context, roomID string, userID string, timeStamp time.Time, message string) (*domain.Message, error) {
 	c, cancel := context.WithTimeout(ctx, u.timeout)
 	defer cancel()
 
 	//message.SentTimestamp, message.FromStudentID, message.RoomID = timeStamp, userID, roomID
 	//get existing message at timestamp
 	existingMessage, err := u.messageRepository.GetMessage(ctx, roomID, timeStamp)
 	if err != nil {
-		return err
+		return nil, errors.NewNotFoundError("Message does not exist")
 	}
 
-	//if userID != existingMessage.FromStudentID{
-	//	return errors.NewUnauthorizedError("Users can only edit their own messages")
-	//}
+	if userID != existingMessage.FromStudentID{
+		return nil, errors.NewUnauthorizedError("Users can only edit their own messages")
+	}
 
 	if message == "" || message == existingMessage.MessageBody {
```

Comment:
should empty or same body be considered an error? I think if it's empty, then the message should be deleted ( in the future might want to ask the user if they want to delete it first ). And if it's the same, then just return the message

Reply:
Yeah makes sense. Changed it.